compiler/src/iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.cpp

```diff
@@ -503,6 +503,133 @@ Flow::wrapOpInDispatchRegion(RewriterBase &rewriter, Operation *op) {
   return newRegionOp;
 }
 
+/// Returns true if the operation is a matmul or matmul-like op.
+bool Flow::isMatmulLikeOp(Operation *op) {
+  auto linalgOp = dyn_cast<linalg::LinalgOp>(op);
+  if (!linalgOp)
+    return false;
+  return linalg::isaContractionOpInterface(linalgOp);
+}
+
+/// Returns true if the operation is an generic op that represents dequant.
+/// This function checks that the genericOp:
+/// 1. Has a body like:
+///      arith.extui
+///      arith.uitofp
+///      arith.subf
+///      arith.mulf
+/// 2. scale, offset and result is f16 or f32, while weight is i4 or i8.
+/// 3. Has 3 parallel dims
+/// 4. Has 2 (weights, scales) or 3 (weights, scales, zero points)
+///    inputs and 1 output
+/// 5. Only weight has same shape as result.
+/// 6. Weight and result have identity indexing map.
+bool Flow::isGroupedDequantizationOp(Operation *op) {
```

Comment:
We can maybe generalize this a bit more.... In my mind requirements are these

1) Only one `ins` operand has a map that is identity. Rest of the operands are projected permutations (and not permutations)
2) The `outs` is of size 1, and has a map that is identity
3) This ins operand with identity map has elementtype that is of smaller bitwidth than the outs operand element type
4) Body has only extsi/mul/add.
5) Extra check can be that all consumers are ops that implement the contraction op interface.